server/source/source.service.ts

```diff
@@ -63,6 +63,10 @@ export class SourceService {
         return this.SourceModel.find({ match }, { _id: 1, href: 1 });
     }
 
+    findHref(href: string) {
```

Comment:
nitpick: The current function name findHref suggests that it returns multiple results, but since we expect a single source, consider renaming it to getSourceByHref for better clarity. Additionally, updating the query to return a single object using .findOne() instead of .find()